packages/runner-shared/src/event-manager.js

```diff
@@ -522,13 +522,13 @@ export const eventManager = {
       Cypress.emit('internal:window:load', { type: 'cross:domain', url })
     })
 
-    Cypress.multiDomainCommunicator.on('expect:domain', (domain) => {
+    Cypress.on('expect:domain', (domain) => {
       localBus.emit('expect:domain', domain)
     })
 
-    Cypress.multiDomainCommunicator.on('before:screenshot', (config) => {
+    Cypress.multiDomainCommunicator.on('before:screenshot', (config, domain) => {
       const callback = () => {
-        Cypress.multiDomainCommunicator.toSpecBridge('before:screenshot:end')
+        Cypress.multiDomainCommunicator.toSpecBridge(domain, 'before:screenshot:end')
```

Comment:
Since we get the originating domain we handily also know where to send the message back